src/collection/account.service.ts

```diff
@@ -354,25 +354,40 @@ export class AccountService {
           );
         const projectName = projectDtoArr[0].collection.name;
         try {
-          this.auditService.recordToolsSync(
-            'start',
-            'unknown',
-            `Start sync: ${service.collection.scmUrl}`,
-            projectName,
-            serviceName,
-          );
-          await this.githubService.refresh(
-            projectName,
-            serviceName,
-            service.collection.scmUrl,
-          );
-          this.auditService.recordToolsSync(
-            'end',
-            'success',
-            `End sync: ${service.collection.scmUrl}`,
-            projectName,
-            serviceName,
-          );
+          const githubRegex = /^https:\/\/github\.com[:/](.+?)\/(.+?)(\.git)?$/;
+          if (
+            service.collection.scmUrl &&
+            service.collection.scmUrl.trim() !== '' &&
+            githubRegex.test(service.collection.scmUrl)
+          ) {
```

Comment:
Check should be outside of try block because only statements that are expected to throw should be in a try.